### PR TITLE
Added profiling scopes around the render dispatch begin/batch/end

### PR DESCRIPTION
**Pull request by JCash**
We want to know if these parts take time as well

### CODE REVIEW

File: engine/render/src/render/render.cpp [hunk 1]
```diff
@@ -721,12 +721,17 @@ namespace dmRender
...
+            DM_PROFILE("Dispatch_Begin");
+
+            // All get begin operation first
+            for (uint32_t i=0;i!=context->m_RenderListDispatch.Size();i++)
+            {
+                const RenderListDispatch& d = context->m_RenderListDispatch[i];
+                params.m_UserData = d.m_UserData;
+                d.m_DispatchFn(params);
+            }
+
```
Comment (JCash) [inline engine/render/src/render/render.cpp:734]: The change looks more than it is.
I've added the scope and indented the code.

File: engine/render/src/render/render.cpp [hunk 2]
```diff
@@ -721,12 +721,17 @@ namespace dmRender
         params.m_Operation = RENDER_LIST_OPERATION_BEGIN;
         params.m_Context = context;
 
-        // All get begin operation first
-        for (uint32_t i=0;i!=context->m_RenderListDispatch.Size();i++)
         {
-            const RenderListDispatch& d = context->m_RenderListDispatch[i];
-            params.m_UserData = d.m_UserData;
-            d.m_DispatchFn(params);
+            DM_PROFILE("Dispatch_Begin");
```
Comment (Jhonnyg) [inline engine/render/src/render/render.cpp:725]: I don't know exactly how the DM_PROFILE generates the scope name, but 'dispatch' is a generic name so maybe we should say what we are dispatching, something like RenderListDispatch?

**Review by Jhonnyg — APPROVED** [review]
Looks good, just a small Q

File: engine/render/src/render/render.cpp (hunk 2 above)
Comment (JCash) [reply engine/render/src/render/render.cpp:725]: The name is local, and it's displayed under the parent scope (currently "Render")

**Pull request merged**